### DISCUSSION TITLE
Change Maps JavaScript API version

### DISCUSSION THREAD

**Discussion by Enr62** [JoomlaEstateAgency/com_jea#38] [Q&A] [answered]
Hi, at the end of June Google sent me this email about a site on a Real Estate Site in Joomla 3 (PHP 8.1):

_“Review which of your Maps JavaScript API resources have incompatible usage. Change your application to avoid a disruption in your service.
Hello Google Maps Customer,
We’re writing to let you know that the Maps JavaScript API will become unavailable to applications using non-compliant libraries, from the Maps JavaScript API version 3.54, releasing to the weekly channel in August 2023.
What do I need to know?
Some third party libraries emulate newer JavaScript functionality, for browsers that may not support them natively. The Maps JavaScript API will start using some newer functionality in the ECMAScript standard, namely Array.from().
While Array.from() is available and working in all browsers supported by the Google Maps JavaScript API, some applications have code which replaces the working version of this with an invalid version from a library - creating an invalid JS environment.
Alternatively, you may specify a version of the Maps JavaScript API lower than v3.54 for the time being, with the caveat that any version is only available for a year. For example, you can specify v=3.53 until April 2024, after which all Maps JavaScript versions will require a fix to your application.”_

While waiting for a total renewal of the site, this second temporary solution (also found on Google Maps Platform) would also be fine, then:

_“An update affected my application
When we update the release channels, it could affect your application. This could occur because your application is relying on undocumented or deprecated features, or because of a bug or bug fix we have introduced.
**As a temporary workaround, you can reconfigure your application to use an older version.**
	1	Open the console in your browser developer tools, and look at the value of google.maps.console
	2	Update the script tag that loads the API, and request an older versione number”_

Now my question is:
if I use the JEA extension (Joomla Estate Agency), where do I find the tag to change the version?
Thanks for the suggestions

**Comment by ilhooq** [comment, marked as answer]
Hi @Enr62 ,

The solution is to make an override of components/com_jea/views/property/tmpl/default_googlemap.php in your template and edit this file.

Find this lines : 

```php
$this->document->addScript(
    'https://maps.google.com/maps/api/js?key=' . $this->params->get('googlemap_api_key') . '&amp;language=' . $lang . '&amp;region=' . $region
);
```
And add the API version like this :

```php
$this->document->addScript(
    'https://maps.google.com/maps/api/js?key=' . 
    $this->params->get('googlemap_api_key') . 
    '&amp;language=' . $lang . '&amp;region=' . $region .
    '&amp;v=3.53' // API version
);
```
